### PR TITLE
Switch from stdweb to websys for #21.

### PR DESCRIPTION
**Pull request by dabreegster**
When I added web support in February, it was easier to get started with
stdweb, since it has the nice cargo web tool. However, stdweb is
unmaintained, winit is deprecating support for it, and the next steps
for web (downloading maps dynamically) have better support for web-sys.

With Alvin's guidance, I got
https://github.com/dabreegster/minimal_websys_winit_glow_demo working
first. This PR cuts A/B Street over too.

I tested abst and the widgetry demo in both native and web. No behavior regressions found!

@michaelkirk

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1,16 +1,11 @@
 # This file is automatically @generated by Cargo.
 # It is not intended for manual editing.
-[[package]]
```
Comment (michaelkirk) [inline Cargo.lock:3]: lots of red! :+1:

File: game/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+            }
+            Err(err) => {
+                timer.error(format!("{} JSON is broken: {}", name, err));
+            }
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+use wasm_bindgen::prelude::*;
```
Comment (michaelkirk) [inline game/src/lib.rs:208]: I'm assuming that other than these wasm32 sections this file was cut/paste from the old game/src/main.rs

File: widgetry/src/backend_glow.rs [hunk 3]
```diff
@@ -353,3 +353,202 @@ impl PrerenderInnards {
...
+///   │4│5│6│
+///   ├─┼─┼─┤
+///   │7│8│9│
+///   └─┴─┴─┘
+///
+/// Texture(0) is reserved for a pure white (no-op) texture.
+///
+/// Implementation is based on the the description of ArrayTextures from:
+/// https://www.khronos.org/opengl/wiki/Array_Texture.
+pub fn load_textures(
```
Comment (michaelkirk) [inline widgetry/src/backend_glow.rs:380]: Nice! Did you have to do anything besides copy this code from native and copy the shader code? Was there anything from native that didn't work on web?

File: widgetry/src/backend_glow_wasm.rs [hunk 4]
```diff
@@ -66,7 +72,7 @@ pub fn setup(window_title: &str) -> (PrerenderInnards, winit::event_loop::EventL
         .collect::<Vec<_>>();
         gl.link_program(program);
         if !gl.get_program_link_status(program) {
-            stdweb::console!(log, "Linking error: %s", gl.get_program_info_log(program));
+            error!("Linking error: {}", gl.get_program_info_log(program));
```
Comment (michaelkirk) [inline widgetry/src/backend_glow_wasm.rs:75]: yay for no-more specialized web-only logging calls!

**Review by michaelkirk — APPROVED** [review]
Nice! Code looks good.

I was able to build and run the widgetry_demo and game.

I did get this warning in the console:
> `[.WebGL-0x7ff15c077800]GL ERROR :GL_INVALID_ENUM : glTexStorage3D: internalFormat was GL_RGBA`

But... I couldn't detect any actual issues with the app. The error seems maybe related to the texture stuff, but I did verify that I could use the "textured" color scheme 🤷:

Screen Shot 2020-10-05 at 6 20 55 PM

Also, I'm getting a crash when hovering over the "game speed" controls. Maybe related to a missing unicode glyph? The crash is in usvg, and error message mentions "←"

Screen Shot 2020-10-05 at 6 24 27 PM

File: game/src/lib.rs (hunk 2 above)
Comment (dabreegster) [reply game/src/lib.rs:208]: Yep. git didn't detect the rename properly. :(

File: widgetry/src/backend_glow.rs (hunk 3 above)
Comment (dabreegster) [reply widgetry/src/backend_glow.rs:380]: Nope, just the straightforward move. `image` compiles fine on wasm, and we're doing the IO with `slurp_file`, which uses the filesystem or `include_dir` underneath.

**Comment by dabreegster** [comment]
> I did get this warning in the console:

I saw this too, but textures also seem to work. Can investigate later.

> Maybe related to a missing unicode glyph?

Ah, I saw this happen in the map loader with Kraków too. I think it's in the text shaping code I hacked in to workaround the lack of rustybuzz. I'll work on that next.

**Pull request merged**